File: spring/auto-testing-example/src/main/java/com/example/autotestingexample/MyController.java
package com.example.autotestingexample;

import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Controller;
import org.springframework.ui.Model;
import org.springframework.web.bind.annotation.GetMapping;

@Controller
@RequiredArgsConstructor
public class MyController {

    private final MyService myService;
    private final MyMapper myMapper;

    @GetMapping("/")
    public String hello(final Model model) {

        model.addAttribute("now", myService.getNow());
        model.addAttribute("boot", myMapper.select());
        return "index";
    }
}

File: thymeleaf/each-for-one/src/main/kotlin/com/example/each_for_one/MyController.kt
package com.example.each_for_one

import org.springframework.stereotype.Controller
import org.springframework.ui.Model
import org.springframework.web.bind.annotation.GetMapping

@Controller
class MyController {

    @GetMapping
    fun index(model: Model): String {
        model.addAttribute("message", MyData("Hello, World!"))
        model.addAttribute("messages", listOf(MyData("Hello, World!"), MyData("Goodbye, World!")))
        return "index"
    }
}

data class MyData(
    val text: String,
)
